src/test/java/codeu/controller/LoginServletTest.java

```diff
@@ -57,8 +58,12 @@ public void setup() {
 
   @Test
   public void testDoGet() throws IOException, ServletException {
```

Comment:
Should you have a test where there is a user already logged in?

Reply:
Good point!

Reply:
After looking back at the code, request.getSession().removeAttribute("user") returns null and executes no matter there is a user attribute in the session or not. So, it think it is not necessary to have separate test cases.